src/realm/owned_data.hpp

```diff
@@ -59,8 +60,16 @@ class OwnedData {
     }
     OwnedData& operator=(const OwnedData& other);
 
-    OwnedData(OwnedData&&) = default;
```

Comment:
These were incorrectly leaving the old object with a null `m_data` and a non-zero `m_size` which is an invalid combination.